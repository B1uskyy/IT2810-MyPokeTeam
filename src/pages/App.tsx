import '../styles/app.css';
import PokemonList from '../components/PokemonList.tsx';
import usePokemons from '../hooks/usePokemons.ts';
import { usePokemonContext } from '../context/usePokemonContext.ts';
import { MdKeyboardArrowRight } from 'react-icons/md';
import { MdKeyboardArrowLeft } from 'react-icons/md';
import { useEffect, useState } from 'react';
import { ListPokemon } from '../interfaces/pokemons.ts';
import Filter from '../components/Filter.tsx';

function App() {
  const { pokemons, fetchNextPage, fetchPrevPage, page, loading, hasMorePokemon } = usePokemons();
  const [filteredPokemons, setFilterPokemons] = useState<ListPokemon[]>([]);
  const [selectedType, setSelectedType] = useState<string>('');
  const { toggleFavorite, isFavorited } = usePokemonContext();

  const pokemonTypes = ['Grass', 'Fire', 'Water', 'Electric'];

  const handleTypeFilterChange = (selectedType: string) => {
    setSelectedType(selectedType);
  };

  useEffect(() => {
    const savedFilter = sessionStorage.getItem('filter');
    if (savedFilter) {
      setSelectedType(savedFilter);
    }
  }, []);

  useEffect(() => {
    if (selectedType) {
      const filtered = pokemons.filter((pokemon) =>
        pokemon.types.some((type: string) => type.toLowerCase() === selectedType.toLowerCase()),
      );
      setFilterPokemons(filtered.slice(0, 20));
    } else {
      setFilterPokemons(pokemons.slice(0, 20));
    }
  }, [pokemons, selectedType]);

  return (
    <>
      <main className="main">
        <Filter name="Type" values={pokemonTypes} onFilterChange={handleTypeFilterChange} />

        {filteredPokemons.length > 0 ? (
          <PokemonList pokemons={filteredPokemons} isFavorited={isFavorited} onToggleFavorite={toggleFavorite} />
        ) : (
          <p>No Pok√©mon of this type in this page</p>
        )}
        {loading && <p>Loading...</p>}

        <section className="button-container">
          <button className="prev" onClick={fetchPrevPage}>
            <MdKeyboardArrowLeft className="arrow" />
            Prev Page
          </button>
          {hasMorePokemon && (
            <button className="next" onClick={fetchNextPage}>
              Next Page
              <MdKeyboardArrowRight className="arrow" />
            </button>
          )}
        </section>
        <h3 className="page-number">Page {page}</h3>
      </main>
    </>
  );
}

export default App;
