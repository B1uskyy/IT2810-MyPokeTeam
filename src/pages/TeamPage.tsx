import '../styles/teampage.css';
import { Banner } from '../components/Banner.tsx';
import { Favorites } from '../components/Favorites.tsx';

export function TeamPage() {
  return (
    <>
      <main className="team-page">
        <Banner />
        <Favorites name="Chris" />
      </main>
    </>
  );
}
