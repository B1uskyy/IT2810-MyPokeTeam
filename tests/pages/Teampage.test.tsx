import { render } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { TeamPage } from '../../src/pages/TeamPage';

describe('Teampage', () => {
  it('matches snapshot with correct props', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/team']}>
        <Routes>
          <Route path={'/team'} element={<TeamPage />} />
        </Routes>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
