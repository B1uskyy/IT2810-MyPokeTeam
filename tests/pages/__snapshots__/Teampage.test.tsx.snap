// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Teampage > matches snapshot with correct props 1`] = `
<DocumentFragment>
  <main
    class="team-page"
  >
    <div
      class="team-wrapper"
    >
      <header>
        <h2
          class="team-name"
        >
          Team name
        </h2>
        <div>
          <button
            class="edit-name-button"
          >
            <svg
              class="edit-name-icon"
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.8995 6.85453L17.1421 11.0972L7.24264 20.9967H3V16.754L12.8995 6.85453ZM14.3137 5.44032L16.435 3.319C16.8256 2.92848 17.4587 2.92848 17.8492 3.319L20.6777 6.14743C21.0682 6.53795 21.0682 7.17112 20.6777 7.56164L18.5563 9.68296L14.3137 5.44032Z"
              />
            </svg>
          </button>
        </div>
      </header>
      <section
        class="team-section"
      >
        <h2>
          No Pok√©mons in your team.
        </h2>
      </section>
    </div>
  </main>
</DocumentFragment>
`;
